crates/block-producer/src/block_builder/mod.rs

```diff
@@ -319,8 +319,11 @@ impl TelemetryInjectorExt for ProposedBlock {
                 .expect("should have less than u32::MAX output notes"),
         );
 
-        let num_batch_created_notes =
-            self.batches().iter().fold(0, |acc, batch| acc + batch.output_notes().len());
+        let num_batch_created_notes = self
+            .batches()
+            .as_slice()
+            .iter()
+            .fold(0, |acc, batch| acc + batch.output_notes().len());
```

Comment:
I would have maybe put this logic into the `OrderedBatches` so that we could do something like this here:
```rust
let num_batch_created_notes = self.batches.num_output_notes();
```

Reply:
Nice suggestion, I've sneaked those in with https://github.com/0xPolygonMiden/miden-node/pull/783.